fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/MetaDataProtoEditorUnitTest.java

```diff
@@ -275,7 +329,131 @@ public void renameOuterTypeWithNestedTypeWithSameName() throws Descriptors.Descr
 
         RecordMetaData metaData = RecordMetaData.build(modifiedFile);
         RecordMetaDataProto.MetaData.Builder metaDataProtoBuilder = metaData.toProto().toBuilder();
-        MetaDataProtoEditor.AmbiguousTypeNameException e = assertThrows(MetaDataProtoEditor.AmbiguousTypeNameException.class, () -> MetaDataProtoEditor.renameRecordType(metaDataProtoBuilder, "OuterRecord", "OtterRecord"));
-        assertEquals("Field middle in message .com.apple.foundationdb.record.test.doublenested.OuterRecord of type OuterRecord might be of type .com.apple.foundationdb.record.test.doublenested.OuterRecord", e.getMessage());
+        MetaDataProtoEditor.renameRecordType(metaDataProtoBuilder, "OuterRecord", "OtterRecord",
+                        getDependencies(metaData));
+        Descriptors.FileDescriptor renamedDescriptor = Descriptors.FileDescriptor.buildFrom(metaDataProtoBuilder.getRecords(), TestRecordsDoubleNestedProto.getDescriptor().getDependencies().toArray(new Descriptors.FileDescriptor[0]));
+        Descriptors.Descriptor renamedOuter = renamedDescriptor.findMessageTypeByName("OtterRecord");
+        Descriptors.Descriptor renamedOuterOuter = renamedOuter.findNestedTypeByName("OuterRecord");
+        assertSame(renamedOuterOuter, renamedOuter.findFieldByName("middle").getMessageType());
+        assertSame(renamedOuterOuter, renamedOuter.findFieldByName("many_middle").getMessageType());
+        assertSame(renamedDescriptor.findMessageTypeByName("OtherRecord"), renamedOuter.findFieldByName("other").getMessageType());
+        Descriptors.Descriptor renamedOuterOuterInner = renamedOuterOuter.findNestedTypeByName("InnerRecord");
+        assertSame(renamedOuterOuterInner, renamedOuterOuter.findFieldByName("inner").getMessageType());
+        assertSame(renamedOuter, renamedOuterOuterInner.findFieldByName("outer").getMessageType());
+
+    }
+
+    public static RecordMetaDataProto.MetaData.Builder loadMetaData(@Nonnull String name) throws IOException {
+        try (@Nullable InputStream input = MetaDataProtoEditorUnitTest.class.getResourceAsStream("/" + name);
+                InputStreamReader reader = new InputStreamReader(Objects.requireNonNull(input,
+                        () -> "No resource: " + name))) {
+            RecordMetaDataProto.MetaData.Builder builder = RecordMetaDataProto.MetaData.newBuilder();
+            JsonFormat.parser().ignoringUnknownFields().merge(reader, builder);
+            return builder;
+        }
+    }
+
+    @ParameterizedTest
+    // Note: Explicitly having the .json here so that you can Cmd+click in the IDE to jump to the file
+    @ValueSource(strings = {"OneBoringType.json",
```

Comment:
It would be good to have a test where renaming one of the types would become another.
It would also be good to have a test where a type just kind of matches the renaming logic.